packages/block-editor/src/components/block-toolbar/utils.js

```diff
@@ -59,6 +59,8 @@ export function useDebouncedShowMovers( {
 		[ isFocused ]
 	);
 
+	useEffect( () => () => clearTimeout( timeoutRef.current ) );
```

Comment:
If you want to uun this on unmount, you need to add `[]` as a second argument of `useEffect` otherwise it runs on each render.

Reply:
Good catch Riad, I see it explained in React docs:

> The default behavior for effects is to fire the effect after every completed render. That way an effect is always recreated if one of its dependencies changes.

> If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. This tells React that your effect doesn’t depend on any values from props or state, so it never needs to re-run. This isn’t handled as a special case — it follows directly from how the dependencies array always works.

https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect

Reply:
Fixed in 02eb218.

Reply:
Good catch, thanks for spotting @youknowriad, and for the fix @gziolo .